### PR TITLE
dns_providers: add HEDNS(Hurricane Electric) support, & Gate aurora & acmedns

### PR DESCRIPTION
**Pull request by menduo**
Thank you for contributing to sewer.                    
Every contribution to sewer is important to us.                       
You may not know it, but you have just contributed to making the world a more safer and secure place.                         

Contributor offers to license certain software (a “Contribution” or multiple
“Contributions”) to sewer, and sewer agrees to accept said Contributions,
under the terms of the MIT License.
Contributor understands and agrees that sewer shall have the irrevocable and perpetual right to make
and distribute copies of any Contribution, as well as to create and distribute collective works and
derivative works of any Contribution, under the MIT License.

Now,                   

## What(What have you changed?)

1. dns_providers: add HEDNS(Hurricane Electric) support.
2. Package Setup: Gate aurora & acmedns

## Why(Why did you change it?)

1. We need HEDNS(Hurricane Electric) support, I use HE DNS for my personal domain.
2. `tldextract`, `apache-libcloud`, `dnspython` are not necessary for people who doesn't use Aurora OR ACME DNS.

Test result:

```bash
$ sewer --dns he --domain acme-test.menduo.net --alt_domains *.acme-test.menduo.net --action run 
chosen_dns_provider. Using he as dns provider.
intialise_success, sewer_version=0.5.5, domain_names=['acme-test.menduo.net', '*.acme-test.menduo.net'], acme_server=https://acme-v02.api...
account key succesfully written to /Users/mdp/ws/contrib/sewer/acme-test.menduo.net.account.key.
acme_register
acme_register_success
apply_for_cert_issuance
apply_for_cert_issuance_success
get_identifier_authorization
get_identifier_authorization_success
create_dns_record start: ('acme-test.menduo.net', 'GvO6EwzDC6DZk1UIlMBT1PSzquQ4qDzQVmlQmYRnDw0')
create_dns_record end: ('acme-test.menduo.net', 'GvO6EwzDC6DZk1UIlMBT1PSzquQ4qDzQVmlQmYRnDw0')
get_identifier_authorization
get_identifier_authorization_success
create_dns_record start: ('*.acme-test.menduo.net', '26bfQC3HXhbJefQ0fi5OG24O8Pr5BAGvgfrmp6fTMro')
create_dns_record end: ('*.acme-test.menduo.net', '26bfQC3HXhbJefQ0fi5OG24O8Pr5BAGvgfrmp6fTMro')
check_authorization_status
check_authorization_status_success
respond_to_challenge
respond_to_challenge_success
check_authorization_status
check_authorization_status_success
respond_to_challenge
respond_to_challenge_success
send_csr
send_csr_success
download_certificate
download_certificate_success
delete_dns_record start: ('acme-test.menduo.net', 'GvO6EwzDC6DZk1UIlMBT1PSzquQ4qDzQVmlQmYRnDw0')
delete_dns_record end: ('acme-test.menduo.net', 'GvO6EwzDC6DZk1UIlMBT1PSzquQ4qDzQVmlQmYRnDw0')
delete_dns_record start: ('*.acme-test.menduo.net', '26bfQC3HXhbJefQ0fi5OG24O8Pr5BAGvgfrmp6fTMro')
delete_dns_record end: ('*.acme-test.menduo.net', '26bfQC3HXhbJefQ0fi5OG24O8Pr5BAGvgfrmp6fTMro')
certificate succesfully written to /Users/mdp/ws/contrib/sewer/acme-test.menduo.net.crt.
certificate key succesfully written to /Users/mdp/ws/contrib/sewer/acme-test.menduo.net.key.
the_end. Certificate Succesfully issued. The certificate, certificate key and account key have been saved in the current directory
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,6 +17,19 @@
 with open(os.path.join(here, 'sewer', '__version__.py'), 'r') as f:
     exec(f.read(), about)
 
+dns_provider_deps_map = {
+    'aliyun': ['aliyun-python-sdk-core-v3', 'aliyun-python-sdk-alidns'],
+    'alicloud': ['aliyun-python-sdk-core-v3', 'aliyun-python-sdk-alidns'],
+    'he': ["hurricanedns"],
+    'hedns': ["hurricanedns"],
```
Comment (komuw) [inline setup.py]: @menduo can we just use either `he` or `hedns` rather than both of them? 
since we already have `cloudflare`, `aurora` I think we should just call it `hurricane`

File: setup.py [hunk 2]
```diff
@@ -17,6 +17,19 @@
 with open(os.path.join(here, 'sewer', '__version__.py'), 'r') as f:
     exec(f.read(), about)
 
+dns_provider_deps_map = {
+    'aliyun': ['aliyun-python-sdk-core-v3', 'aliyun-python-sdk-alidns'],
+    'alicloud': ['aliyun-python-sdk-core-v3', 'aliyun-python-sdk-alidns'],
```
Comment (komuw) [inline setup.py]: @menduo I think we should probably also just use either `aliyun` or `alicloud` rather than have them as different. 
But I'm not sure, I've never worked with aliyun or alicloud so I do not know how different they are. 
I'll leave you to make that judgement

File: setup.py [hunk 3]
```diff
@@ -17,6 +17,19 @@
...
+    'he': ["hurricanedns"],
+    'hedns': ["hurricanedns"],
+    'aurora': ["tldextract", "apache-libcloud"],
+    'acmedns': ["dnspython", ],
+}
```
Comment (komuw) [inline setup.py]: Thanks for separating/gating `aurora` and `acmeDns`, I've been wanting to do that for sometime

File: sewer/cli.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ def main():
         "--dns",
         type=str,
         required=True,
-        choices=['cloudflare', 'aurora', 'acmedns', "aliyun", "alicloud"],
+        choices=['cloudflare', 'aurora', 'acmedns', "aliyun", "alicloud", "he", "hedns"],
```
Comment (komuw) [inline sewer/cli.py]: @menduo we should use just one `he` or `hedns` 
since we already have `cloudflare`, `aurora` I think we should just call it `hurricane`

File: sewer/dns_providers/__init__.py [hunk 5]
```diff
@@ -3,3 +3,4 @@
 from .cloudflare import CloudFlareDns  # noqa: F401
 from .acmedns import AcmeDnsDns  # noqa: F401
 from .aliyundns import AliyunDNS  # noqa: F401
+from .hedns import HEDNS  # noqa: F401
```
Comment (komuw) [inline sewer/dns_providers/__init__.py]: @menduo The other classes are called `CloudFlareDns` `AcmeDnsDns` 
so I think this should also be called `HeDns` so that we can have consistency. 

We should probably rename `AliyunDNS` to `AliyunDns` for the same reasons; but I do not know if that will break compatibility for you. If it would break compatibility, maybe we can do it in another Pull request rather than this one

File: setup.py (hunk 1 above)
Comment (menduo) [reply setup.py]: OK, I agree.

File: sewer/dns_providers/__init__.py (hunk 5 above)
Comment (menduo) [reply sewer/dns_providers/__init__.py]: OK.
